### PR TITLE
feat: add extended numeric types (Float, BigInt, BigRat, FixedPoint)

### PR DESCRIPTION
**Pull request by spreston8**
## Summary

Implements the Rholang numeric types spec (Michael Stay, 2025-11-13), adding four new numeric types to the VM, compiler, and runtime:

- **Float** (`f64`) — IEEE 754 double precision, suffix `f32`/`f64`. NaN != NaN, MOD rejected per spec.
- **BigInt** — arbitrary-precision signed integers, suffix `n`. No size cap — gas scales with operand size.
- **BigRat** — exact rationals as ratio of BigInts, suffix `r`. MOD always returns 0. Auto-reduced to lowest terms via GCD (`num_rational::BigRational`).
- **FixedPoint** — shifted BigInt with fixed decimal scale, suffix `p<scale>`. Scale-matching required for all ops. Scale-preserving multiplication with floor division. Shifted division. C99 identity `(a/b)*b + a%b == a`.

No implicit type coercion — all binary ops require matching types. Cross-type operations produce a type mismatch error.

### Key changes
- **Value enum** (`rholang-rspace`): 4 new variants, custom `PartialEq` (NaN semantics), `PartialOrd`, `Display`
- **VM execute** (`rholang-vm`): all arithmetic (ADD/SUB/MUL/DIV/MOD/NEG) and comparison opcodes extended for all types
- **Compiler codegen** (`rholang-compiler`): constant pool (`PUSH_CONST` opcode) for values that don't fit in i16 immediates, compile functions for all 6 literal types
- **Render simplification**: 4 duplicated `render_value` functions replaced with delegation to `Value::Display`

### Review feedback addressed
- Removed BigInt size cap entirely (gas is the rate limiter, not an arbitrary byte limit)
- Fixed FixedPoint multiplication: scale-preserving with floor division instead of scale-doubling
- Added signed int (`i64`) division-by-zero tests
- BigRat always in lowest terms (GCD auto-reduction via `num_rational::BigRational`)

### Open question
- FixedPoint multiplication rounding mode: currently floor division (toward -infinity). Awaiting confirmation on whether truncation toward zero or unbiased (banker's) rounding is preferred.

### Design doc reference
- `Numeric Types.md` (Michael Stay, 2025-11-13)

### Companion PR
- [f1r3node-rust PR #425](https://github.com/F1R3FLY-io/f1r3node/pull/425) — same numeric types implemented in the f1r3node-rust interpreter

## Test plan
- [x] 16 VM-level tests in `rholang-vm/tests/numeric_types_tests.rs` (arithmetic, edge cases, errors, cross-type, int div-by-zero)
- [x] 7 end-to-end compiler tests in `rholang-compiler/tests/numeric_types.rs` (parse -> compile -> execute)
- [x] Unit tests for Value equality, ordering, Display in `rholang-rspace/src/value.rs`
- [x] All pre-existing tests pass
- [x] `cargo test -p rholang-vm -p rholang-compiler -p rholang-rspace` passes with 0 failures
- [x] `cargo clippy --all-targets -- -D warnings` clean

### CODE REVIEW

File: rholang-vm/tests/numeric_types_tests.rs [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+fn test_fixedpoint_add_sub() {
+    // Same-scale required; 1.50 + 2.25 = 3.75
+    assert_eq!(binop(fixed(150, 2), fixed(225, 2), Opcode::ADD).unwrap(), fixed(375, 2));
+    // 5.00 - 3.25 = 1.75
+    assert_eq!(binop(fixed(500, 2), fixed(325, 2), Opcode::SUB).unwrap(), fixed(175, 2));
+}
+
+#[test]
+fn test_fixedpoint_mul_doubles_scale() {
+    // Spec: scale doubles on multiplication. 1.5p1 * 2.0p1 = 3.00p2
```
Comment (metaweta) [inline rholang-vm/tests/numeric_types_tests.rs]: No, `1.5p1 * 2.0p1 = 3.0p1`.  They can lose precision: `0.1p1 * 0.1p1 = 0.0p1` since `0.01` is unrepresentable as a `p1`.  I suppose we should have a discussion about whether we need to do [unbiased rounding](https://www.embeddedrelated.com/showarticle/1015.php#:~:text=BIASED%20VS.%20UNBIASED%20ROUNDING) when multiplying.

File: rholang-vm/tests/numeric_types_tests.rs [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+#[test]
+fn test_fixedpoint_neg_and_errors() {
+    assert_eq!(unaryop(fixed(150, 2), Opcode::NEG).unwrap(), fixed(-150, 2));
+    // Different scales -> type mismatch
+    assert!(binop_err(fixed(150, 2), fixed(15, 1), Opcode::ADD).contains("type mismatch"));
+    // Division by zero
+    assert!(binop_err(fixed(100, 1), fixed(0, 1), Opcode::DIV).contains("division by zero"));
+    assert!(binop_err(fixed(100, 1), fixed(0, 1), Opcode::MOD).contains("division by zero"));
+}
```
[the remark this answers is not in the document]
Comment (metaweta) [reply rholang-vm/tests/numeric_types_tests.rs]: BTW overflow and underflow for bigints need to consider multiplication as well as addition.  And I don't see the implementation of multiplication for bigrats, but it seems plausible that the algorithm might lose precision if done naively (i.e. `(a/b) * (c/d) = (ac/bd)`).

**Review by berecik — APPROVED** [review]

**Review by metaweta — APPROVED** [review]
